### PR TITLE
Quantize, dequantize optimizations for HiFi targets

### PR DESCRIPTION
**Pull request by cad-audio**
BUG=none

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/quantize.cc [hunk 1]
```diff
@@ -135,6 +151,17 @@ TfLiteStatus EvalXtensa(TfLiteContext* context, TfLiteNode* node) {
...
+          TF_LITE_ENSURE_EQ(
+              context,
+              xa_nn_elm_requantize_asym16s_asym8s(
+                  tflite::micro::GetTensorData<int8_t>(output),
+                  tflite::micro::GetTensorData<int16_t>(input),
+                  op_data->input_zero_point, op_data->quantization_params.zero_point,
+                  op_data->requantize_output_shift,
+                  op_data->requantize_output_multiplier, size),
+              0);
+#else // defined(HIFI5) || defined(HIFI4) || defined(HIFI4_INTERNAL)
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/quantize.cc:164]: This #else codeblock is the same as the #if. It seems like this code could have just been left as is?

File: tensorflow/lite/micro/kernels/xtensa/quantize.cc [hunk 2]
```diff
@@ -135,6 +151,17 @@ TfLiteStatus EvalXtensa(TfLiteContext* context, TfLiteNode* node) {
       switch (output->type) {
         case kTfLiteInt8: {
           int size = ElementCount(*input->dims);
+#if defined(HIFI5) || defined(HIFI4) || defined(HIFI4_INTERNAL)
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/quantize.cc:154]: Note to self: Remove the HIFI4_INTERNAL conditional, that is deprecated.

File: tensorflow/lite/micro/kernels/xtensa/quantize.cc (hunk 1 above)
Comment (rascani) [reply tensorflow/lite/micro/kernels/xtensa/quantize.cc:164]: else block should fall back to reference op

Comment (rascani) [reply tensorflow/lite/micro/kernels/xtensa/quantize.cc:164]: Actually, looks like none of these #ifdefs are needed. In the Eval function, we already do a check for HIFI3, 4, & 5.  If those aren't defined, we'll use a reference implementation of Eval. I think the only #ifdef check still needed is the HIFI_VFPU for the float quantizations. I'll add a commit to this effect. Please let me know if I get anything wrong.

**Review by rascani — APPROVED** [review]

**Comment by rascani** [comment]
@Suyash20012000 - Can you sign the CLA? https://github.com/tensorflow/tflite-micro/pull/2544/checks?check_run_id=23948620573

**Comment by Suyash20012000** [comment]
@rascani I have signed the CLA. Requesting you to check.

**Pull request merged**